contrib/babelfishpg_tsql/src/pltsql.h

```diff
@@ -2175,7 +2175,7 @@ extern void update_ViewStmt(Node *n, const char *view_schema);
 extern void update_AlterDefaultPrivilegesStmt(Node *n, const char *schema, const char *role1, const char *role2, const char *grantee, const char *priv);
 extern AccessPriv *make_accesspriv_node(const char *priv_name);
 extern RoleSpec   *make_rolespec_node(const char *rolename);
-extern void pltsql_check_or_set_default_typmod(TypeName *typeName, int32 *typmod, bool is_cast);
+extern void pltsql_check_or_set_default_typmod(TypeName *typeName, int32 *typmod, bool is_cast, bool is_procedure_or_func);
```

Comment:
From engine side we are handling functionalities like create/alter table , create type, etc. using the hook check_or_set_default_typmod_hook which internally calls the function pltsql_check_or_set_default_typmod. By keeping the value for is_procedure_or_func as false we make sure to keep these functionalities same as before, as we target to only update proc/func typmod values through this.

I have named the variable to be is_procedure_or_func because it handles both the procedure and function arguments together as both expect the same behaviour from pltsql_check_or_set_default_typmod function. Thus it could be either a procedure or function to be true.

Reply:
we do not need to introduce new field for that. You can refactor pltsql_check_or_set_default_typmod using some helper. We can directly call this directly from extension with is_procedure_or_func = true and it will be false if helper gets called through hook.

Reply:
Done